### PR TITLE
Sprint 1 PR

### PR DESCRIPTION
**Pull request by emil0519**
[Github Label List](https://emil-w0p3.web.app/)
[Storybook for Sprint 1](https://emil0519storybooks.web.app/?path=/story/github-repo--component/)
[Presentation slides](https://www.canva.com/design/DAFNSNCCxvo/S91-zFg77EjdsIOD6ZgKhA/view?utm_content=DAFNSNCCxvo&utm_campaign=designshare&utm_medium=link2&utm_source=sharebutton)

In this Sprint, I have completed:
1. Layout of Github Label List
2. API to get, create, update and delete labels
3. Random color picker and detect clicking event within and outside specified component(s).
4. Storybook component(sharing components only).

### CODE REVIEW

File: src/component/BoxHeader.tsx [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+  }, []);
+  useEffect(() => {
+    setLabel(updatedLabels);
+    // 每次有新label的時候會re-render一次
+  }, [updatedLabels]);
+  useEffect(() => {
+    if (typeof labels === "string") {
+      setLabel(JSON.parse(labels));
+    }
+  }, []);
```
Comment (max80713) [inline src/component/BoxHeader.tsx:24]: labels 初始值必為 undefined 不需要 parse 然後又呼叫 setLabels

File: src/component/ColorBricks.tsx [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+    useComponentVisible(false);
+  const [editOpen, setEditOpen] = useState(false);
+
+  const handleClickOutside = () => {
+    setEditOpen(false);
+    setIsComponentVisible(false);
+  };
+  useOnClickOutside(ref, handleClickOutside);
+
+  const solidColorList: any = [
```
Comment (max80713) [inline src/component/ColorBricks.tsx:16]: 避免 any

File: src/component/ColorBricks.tsx [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+    },
+    {
+      name: "#0052c8",
+    },
+    {
+      name: "#521be2",
+    },
+  ];
+
+  const lightColorList: any = [
```
Comment (max80713) [inline src/component/ColorBricks.tsx:43]: 避免 any

File: src/component/ColorBricks.tsx [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+            color: e.target.value,
+          });
+        }}
+        onClick={() => {
+          setIsComponentVisible(true);
+          setEditOpen(true);
+        }}
+      ></ColorInput>
+      {isComponentVisible && (
+        <ColorSelector style={{ display: "flex" }} ref={ref}>
```
Comment (max80713) [inline src/component/ColorBricks.tsx:89]: 避免 inline style

File: src/component/LabelList.tsx [hunk 5]
```diff
@@ -0,0 +1,753 @@
...
+    }
+  }, [updatedLabels]);
+
+  if (label === undefined || updateLabelInfo === undefined) {
+    return (
+      <>
+        <NoLabelWrapper>
+          <NoLabelText>Loading</NoLabelText>
+        </NoLabelWrapper>
+      </>
```
Comment (max80713) [inline src/component/LabelList.tsx]: redundant <></>

**Comment by emil0519** [comment]
Thanks for the comment.  Have resolved most but not all throughout the week, will resolve others later.

**Pull request closed without merging**